homeassistant/components/freebox/camera.py

```diff
@@ -0,0 +1,191 @@
+        )
+        self._rstp = self.get_command_id(node["show_endpoints"], ATTR_RTSP)
+        self._disk = self.get_command_id(node["show_endpoints"], ATTR_DISK)
+
+        self.update_node(node)
+
+    @property
+    def flip(self) -> bool:
+        """Return flip."""
+        return self._flip
+
+    async def async_flip(self, entity: FreeboxCamera) -> None:
+        """Flip the camera stream."""
+        self._flip = not entity.flip
+        await entity.set_home_endpoint_value(entity._command_flip, entity.flip)
+
+    async def async_enable_motion_detection(self) -> None:
+        """Enable motion detection in the camera."""
+        await self.set_home_endpoint_value(self._command_motion_detection, True)
+        self._attr_motion_detection_enabled = True
+
+    async def async_disable_motion_detection(self) -> None:
+        """Disable motion detection in camera."""
+        await self.set_home_endpoint_value(self._command_motion_detection, False)
+        self._attr_motion_detection_enabled = False
+
+    async def async_update_signal(self) -> None:
+        """Update the camera node."""
+        self.update_node(self._router.home_devices[self._id])
+
+    def update_node(self, node):
+        """Update params."""
+        self._name = node["label"].strip()
+
+        # Get status
+        if self._node["status"] == "active":
+            self._attr_is_streaming = True
+        else:
+            self._attr_is_streaming = False
+        # Parse all endpoints values & needed commands
+        for endpoint in filter(
+            lambda x: (x["ep_type"] == "signal"), node["show_endpoints"]
+        ):
+            if endpoint["name"] == ATTR_DETECTION:
+                self._attr_motion_detection_enabled = endpoint["value"]
+            elif endpoint["name"] == ATTR_ACTIVATION:
+                self._activation_with_alarm = endpoint["value"]
+            elif endpoint["name"] == ATTR_QUALITY:
+                self._high_quality_video = endpoint["value"]
+            elif endpoint["name"] == ATTR_SENSITIVITY:
+                self._motion_sensitivity = endpoint["value"]
+            elif endpoint["name"] == ATTR_THRESHOLD:
+                self._motion_threshold = endpoint["value"]
+            elif endpoint["name"] == ATTR_FLIP:
+                self._flip = endpoint["value"]
+            elif endpoint["name"] == ATTR_TIMESTAMP:
+                self._timestamp = endpoint["value"]
+            elif endpoint["name"] == ATTR_VOLUME:
+                self._volume_micro = endpoint["value"]
+            elif endpoint["name"] == ATTR_SOUND_DETECTION:
+                self._sound_detection = endpoint["value"]
+            elif endpoint["name"] == ATTR_SOUND_TRIGGER:
+                self._sound_trigger = endpoint["value"]
+            elif endpoint["name"] == ATTR_RTSP:
+                self._rtsp = endpoint["value"]
+            elif endpoint["name"] == ATTR_DISK:
+                self._disk = endpoint["value"]
```

Comment:
Should we use `get_value` from home_base.py removed on commit https://github.com/home-assistant/core/pull/88104/commits/f83dfaec5931e97338e598bbad40cfb9c1727563 here ?

May be cleaner, but don't know about perf.

Reply:
hum... good question.... get_value was used by the other devices, not for cameras.... 
I'll check it's better call 12 times get_value function or lopping for all show_endpoints...

Reply:
In my  RPI, looping is faster than get_value but with get_value the code is easier to read......  

2023-03-18 07:22:04.740 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -7.9e-05
2023-03-18 07:22:04.741 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.4e-05
2023-03-18 07:22:34.825 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.1e-05
2023-03-18 07:22:34.826 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.3e-05
2023-03-18 07:23:04.755 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -0.000207
2023-03-18 07:23:04.756 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.5e-05
2023-03-18 07:23:34.761 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -7.7e-05
2023-03-18 07:23:34.763 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -0.000202
2023-03-18 07:24:04.752 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.4e-05
2023-03-18 07:24:04.753 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.7e-05
2023-03-18 07:24:34.759 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -7.7e-05
2023-03-18 07:24:34.760 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -7.9e-05
2023-03-18 07:25:04.760 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.7e-05
2023-03-18 07:25:04.761 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.3e-05
2023-03-18 07:25:34.761 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -7.4e-05
2023-03-18 07:25:34.762 WARNING (MainThread) [homeassistant.components.freebox.camera] Loop vs Get -8.2e-05

Reply:
`get_value` used, fixed